### PR TITLE
MEGANAV lives!! Adds new SiteNavigation component.

### PR DESCRIPTION
**Pull request by weerd**
### What does this PR do?

Kapture 2019-10-02 at 17 36 14

This PR adds the initial structure for the new site navigation as a React component called `SiteNavigation`, and adds the initial styles for the new navigation to work responsively. Currently this cannot be tested anywhere so I added the GIF above to showcase interaction functionality and a bunch of screen shots to help provide context for the code.

_Note:_ The dropdown currently has very minimal styles added to it, mostly just layout on the grid, and some heading styles. More to come on that front.

### Any background context you want to provide?

- [Small](https://user-images.githubusercontent.com/105849/66079016-3bbe5b80-e531-11e9-9bfb-8793ee5fbaeb.png)
- [Small with dropdown](https://user-images.githubusercontent.com/105849/66079055-5395df80-e531-11e9-9e7b-a573225d1933.png)
- [Medium](https://user-images.githubusercontent.com/105849/66079072-5abced80-e531-11e9-837a-407edbf8f4f9.png)
- [Medium with dropdown](https://user-images.githubusercontent.com/105849/66079094-65778280-e531-11e9-84f5-c844c81b7f38.png)
- [Large](https://user-images.githubusercontent.com/105849/66079119-71634480-e531-11e9-999e-a242b6c409fb.png)
- [Large with dropdown](https://user-images.githubusercontent.com/105849/66079139-7cb67000-e531-11e9-8769-8586fc52eaed.png)

### What are the relevant tickets/cards?

Refs [Pivotal ID #168552873](https://www.pivotaltracker.com/story/show/168552873)

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [x] Added screenshot to PR description of related front-end updates on **small** screens.
* [x] Added screenshot to PR description of related front-end updates on **medium** screens.
* [x] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by weerd** [comment]
I'd like to eventually convert the styles to use utility classes, but since we don't have Tailwind fully implemented in Phoenix and the styles for this SiteNavigation are pretty complex, it didn't make sense to try and use utility classes without having the responsive classes that Tailwind provides.

**Comment by lkpttn** [comment]
Looks good! We'll sync up on dropdown styles later

File: resources/assets/components/utilities/SiteNavigation/SiteNavigation.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            </section>
+
+            <section className="main-subnav__featured menu-subnav__content">
+              <img
+                className="mb-4"
+                src="http://placekitten.com/g/550/250"
+                alt="temporary place kitten"
+              />
+              <h1>Take Back the Kittens</h1>
+              <p>Donec ullamcorper nulla non metus auctor fringilla.</p>
```
Comment (Vmack) [inline resources/assets/components/utilities/SiteNavigation/SiteNavigation.js]: Everything looks good just scratching my head on this line. Was this supposed to be included?

Comment (weerd) [reply resources/assets/components/utilities/SiteNavigation/SiteNavigation.js]: Oh yeah! It's just sample content I added so I had something to work with. This navigation currently won't show _anywhere_ on the website, so it's safe for now to keep it, haha! I probably should have clarified that 😬

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    position: fixed;
+    top: 0;
+    width: 100%;
+    z-index: -1;
+  }
+}
+
+.menu-subnav__links {
+  > h1 {
+    a {
```
Comment (DFurnes) [inline resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: Nested selectors like this (`.menu-subnav__links > h1 a`) make me a little nervous because of how specific they are. Should we make a class for these elements?

File: resources/assets/components/utilities/SiteNavigation/SiteNavigation.js (hunk 1 above)
Comment (Vmack) [reply resources/assets/components/utilities/SiteNavigation/SiteNavigation.js]: Oh great! No worries, I wanted to make sure it was just a test. You're good to good to go!

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+.main-subnav__links-causes {
+  grid-column: full-start / full-end;
+
+  @include media($large) {
+    grid-column: wide-start / span 2;
+  }
+}
+
+.main-subnav__links-campaigns {
```
Comment (DFurnes) [inline resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: Is there a common "thing" that describes both these (`.main-subnav__links-causes` and `.main-subnav__links-campaigns`) so they could be the same selector? Maybe `.main-subnav__full`?

File: resources/assets/components/utilities/SiteNavigation/SiteNavigation.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+import './site-navigation.scss';
+import dsLogo from '../../../images/ds_logo.svg';
+import searchIcon from '../../../images/search_icon.svg';
```
Comment (DFurnes) [inline resources/assets/components/utilities/SiteNavigation/SiteNavigation.js]: Let's put these in with the component so that we don't have to jump up so many directories `../../..`.

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss [hunk 5]
```diff
@@ -0,0 +1,219 @@
+@import '../../../scss/next-toolbox.scss';
+
+$spacing_1x: 16px;
+$spacing_1_5x: 24px;
+$spacing_2x: 32px;
```
Comment (DFurnes) [inline resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: Are these variable names Tailwind-inspired?

File: resources/assets/scss/base.scss [hunk 6]
```diff
@@ -678,6 +686,10 @@ p {
   font-size: $font-hero;
 }
 
+.visible {
+  visibility: visible;
+}
+
```
Comment (weerd) [inline resources/assets/scss/base.scss]: Added these in when I initially started to try and use utility classes for the new navigation, but then realized it was not going to be ideal without the Tailwind responsive classes. I left these in though, since they seemed helpful!

**Comment by weerd** [comment]
One quick additional note! I still need to do some browser testing to see how this all works out in MS Edge browser 😬

**Comment by DFurnes** [comment]
Just dropping this based on our conversation yesterday – it'd be nice to give this a round of "polish" before we launch the feature to [be more forgiving when users cross out of the "hover" element](https://css-tricks.com/menus-with-dynamic-hit-areas/), since that's a common UX problem with mega-navs like this.

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+  padding: $spacing_1x 0 0;
+  z-index: 500; // @TODO: create variable in next-toolbox.scss
+
+  @include media($medium) {
+    padding: $spacing_1_5x $spacing_2x 0;
+  }
+}
+
+.menu-nav__item {
+  &:hover {
```
Comment (DFurnes) [inline resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: How does this work for users on touch screens, like iPads? (Honestly not sure!)

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss (hunk 2 above)
Comment (weerd) [reply resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: Oh ya, I meant these more as temporary, since I was only doing simple styling for the subnav for links for now. More to come on the styling for the stuff _inside_ the dropdowns!

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss (hunk 3 above)
Comment (weerd) [reply resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: Yep, potentially. I think for general styling I want to end up combining them and will happen with the styling for the content inside the dropdown as mentioned in this [comment](https://github.com/DoSomething/phoenix-next/pull/1618#discussion_r331163968), buuut I think we'll still need classes like this to help define their specific grid layout rules (although there may be a way around it).

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss (hunk 7 above)
Comment (weerd) [reply resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: It doesn't! Haha, but defs the intention for now. This was just the main CSS pass. For touch screens I need the JS pass to help with this which is for follow-up PRs 💃

File: resources/assets/components/utilities/SiteNavigation/SiteNavigation.js (hunk 4 above)
Comment (weerd) [reply resources/assets/components/utilities/SiteNavigation/SiteNavigation.js]: Just wondering if (in particular for the logo) if it make sense to keep it in a more accessible directory for other components that may need it. The search seems like it could be used in other components too which was my initial reasoning with keeping it in a more generic directory outside the component. Thoughts?

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss (hunk 2 above)
Comment (weerd) [reply resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: I should have clarified better that the dropdown styles were very basic and not completed. Aim for this PR was getting the main nav functional and responsively styled!

Comment (DFurnes) [reply resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: Alrighty!

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss (hunk 3 above)
Comment (DFurnes) [reply resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: H'okay!

File: resources/assets/components/utilities/SiteNavigation/SiteNavigation.js (hunk 4 above)
Comment (DFurnes) [reply resources/assets/components/utilities/SiteNavigation/SiteNavigation.js]: If so, let's make a re-usable component for those elements (e.g. `<Logo>` or <Search>`) and keep the referenced assets alongside the component they're used by.

File: resources/assets/components/utilities/SiteNavigation/site-navigation.scss (hunk 5 above)
Comment (DFurnes) [reply resources/assets/components/utilities/SiteNavigation/site-navigation.scss]: Gotcha. 👍

**Comment by weerd** [comment]
Thanks for the CSS-Tricks link @DFurnes! Will be super useful! 🙌

File: resources/assets/components/utilities/SiteNavigation/SiteNavigation.js (hunk 4 above)
Comment (weerd) [reply resources/assets/components/utilities/SiteNavigation/SiteNavigation.js]: Chatted w/ @DFurnes and gonna take his suggested approach in upcoming PR 😄
